### DISCUSSION TITLE
Doubts about creating plugins

### DISCUSSION THREAD

**Discussion by denerFernandes** [saleor/saleor#8393] [General]
Dzień dobry,

Guys i want to develop a new plugin to implement a headless cms functions on Saleor, but a have doubts about this.

- Can migrations be at plugin level? i can I have specific migrations for the plugin to create a new tables on database?
- Can the plugin create new items and pages in admin? 
- About Graphql can the plugin create queries and mutations?
- Have documentation to this? (If have, I don't found, i'm sorry! )

Thanks!

**Comment by jaydotcom** [comment]
Hi there, I was trying to create a plugin for my need.
As Fernandes, I need to create new page in admin/dashboard, but I haven't found any documentation  for this !
Best regards

**Comment by korycins** [comment]
Hey,
The plugin doesn't allow you to extend dashboard panels. I would recommend building an App for this, this is a separate instance that communicates with Saleor via API and receives updates from Saleor via webhooks.
The app can also extend a dashboard by mounting its own iframe views. 
App docs: https://docs.saleor.io/docs/3.0/developer/extending/apps/key-concepts
Our framework for building apps. You can find there app samples: https://github.com/saleor/saleor-app-framework-python

The one thing that I believe is missing in docs is that when you're having an app manifest you can install it directly from dashboard:
http://<dashboard-url>/apps/install?manifestUrl=https://<app-domain>/configuration/manifest

**Reply by jaydotcom** [reply]
Thanks for this, was really not clear on my mind ;)

**Comment by jaydotcom** [comment]
Dear @korycins 
I've been further your reading and in fact it seems to be the right way to manage my problem.

But I'm now stucked a bit further : I've created a manifest following your guidelines here : https://docs.saleor.io/docs/3.0/developer/extending/apps/manifest

Issue is when installing I got the following error : ['Missing required fields for app extension: type, view.']

I went down the code and in fact there is a list of required fields containing these fields : 
![image](https://user-images.githubusercontent.com/15248619/154252158-bd9ca3f0-6750-47fb-90dc-7f31465920a7.png)

But I can't find what values I need to put inside following your guideline on app extension: https://docs.saleor.io/docs/3.0/developer/extending/apps/extensions#key-concepts

Best regards

**Reply by korycins** [reply]
Hey, 
Are you updated to the newest main/alfa release (https://github.com/saleor/saleor/releases/tag/3.1.0-a.25). 
This should solve your issue.

**Reply by jaydotcom** [reply]
Thanks @korycins , Now installed !
But at the end even after activation I can't access anything in my dashboard.

Here is my manifest : https://dagoapi-dev.herokuapp.com/api/vendors/6202b85b0883a9c88323d309/manifest

I've activated it on my instance : 
![image](https://user-images.githubusercontent.com/15248619/154806499-0c80aaaf-6d6d-451d-8699-a3261c613936.png)

Was thinking I'll get something under "clients" menu but nothing appears : 
![image](https://user-images.githubusercontent.com/15248619/154806522-3d94d7d3-fbd5-4f59-afc7-de9966cdaac8.png)

And if I click on "Open app" in menu, nothing happens : 
![image](https://user-images.githubusercontent.com/15248619/154806542-957d1a51-1982-4cdc-bce9-ff3586f4b3a6.png)

It's supposed to open an iframe in my sandbox app here : https://dago-saleor-app-getstream-dev.herokuapp.com/?vendorId=6202b85b0883a9c88323d309 (Loading can be long => Host on a free Heroku instance => Instance needs to launch).

Any help ?

Best regards

**Reply by korycins** [reply]
Hey @jaydotcom, 

Can you copy and paste a manifest here? The manifest url doesn't work.

**Reply by jaydotcom** [reply]
Sorry, dunno why was not working anymore.
Here we go : 
`{"id":"dago.saleor.getstream","version":"1.0.0","name":"Dago Chat","permissions":["MANAGE_USERS","MANAGE_ORDERS"],"appUrl":"https://dago-saleor-app-getstream-dev.herokuapp.com/?vendorId=6202b85b0883a9c88323d309","tokenTargetUrl":"https://dagoapi-dev.herokuapp.com/api/vendors/6202b85b0883a9c88323d309/manifest/token","extensions":[{"label":"Messages","mount":"NAVIGATION_CUSTOMERS","target":"POPUP","permissions":["MANAGE_USERS","MANAGE_ORDERS"],"url":"https://dago-saleor-app-getstream-dev.herokuapp.com/?vendorId=6202b85b0883a9c88323d309"}]}`

**Reply by korycins** [reply]
@jaydotcom 
Hhm, the manifest looks okay. You need to debug it somehow.
Make sure, that you're using the newest version of dashboard and core. 
Here you have my custom build for testing apps: https://github.com/korycins/saleor-app-framework-python/tree/test_heroku_build (it is fork of our framework for apps). You can push it on heroku, it works for me. Manifest can be found:  `<url>/configuration/manifest`

**Reply by jaydotcom** [reply]
I've updated all, clear my browser cache and seems ok now.
Sorry for this and thanks for your help
